lib/ulid/rails/type.rb

```diff
@@ -26,7 +26,13 @@ def cast(value)
       def serialize(value)
         return if value.nil?
 
-        case ActiveRecord::Base.connection_config[:adapter]
+        if ::ActiveRecord::Base.respond_to?(:connection_db_config)
+          adapter = ::ActiveRecord::Base.connection_db_config.configuration_hash[:adapter]
+        else
+          adapter = ::ActiveRecord::Base.connection_config[:adapter]
+        end
```

Comment:
Can we change this to a private `#adapter` method instead?
```suggestion
def adapter
  if ::ActiveRecord::Base.respond_to?(:connection_db_config)
    ::ActiveRecord::Base.connection_db_config.configuration_hash[:adapter]
  else
    ::ActiveRecord::Base.connection_config[:adapter]
  end
end
```